editor/components/inner-blocks/index.js

```diff
@@ -22,23 +22,23 @@ import BlockList from '../block-list';
 import { withBlockEditContext } from '../block-edit/context';
 
 class InnerBlocks extends Component {
-	componentDidMount() {
+	constructor() {
+		super( ...arguments );
+
 		this.updateNestedSettings();
+	}
+
+	componentDidMount() {
 		this.synchronizeBlocksWithTemplate();
 	}
 
 	componentDidUpdate( prevProps ) {
-		const { template, block } = this.props;
+		const { template } = this.props;
 
 		this.updateNestedSettings();
 
 		const hasTemplateChanged = ! isEqual( template, prevProps.template );
```

Comment:
Here we compare using equal while on allowedBlocks we compare by reference. We should probably decide on one of the approaches while comparing by reference is more performant this comparison is more developer friendly.

Reply:
I agree we need consistency here. Not sure performance is a fair measurement in this case though, since in almost all cases `allowedBlocks` will be referentially unequal, since with usage like `<InnerBlocks allowed={ [ 'core/paragraph' ] } />` we are passing a new array reference each time.

Reply:
Our docs already contain the samples using a constant.

const ALLOWED_BLOCKS = [ 'core/paragraph' ];
...
<InnerBlocks allowed={ ALLOWED_BLOCKS } />

For dynamic cases, we can save a reference in the state.
But yes is more complex for developers and we may get rerenders on external blocks.

Reply:
I think we may be better off doing a deep equality check on the allowed blocks. It's not too much more effort to dive into the array of strings, and avoids a much more concerning overhead if the developer doesn't assign allowed blocks as a constant reference (which itself is overhead in its own right). I'll plan to open a separate pull request proposing this.